packages/android/barnard/src/main/kotlin/org/levarac/barnard/BarnardParticipantRelay.kt

```
@@ -0,0 +1,111 @@
+        }
+        if (candidates.size == 32) return BarnardRelayObservationResult.SATURATED
+        val verified = verifier.verify(envelope.copyOf(), current) as? BarnardRelayVerification.RegistryVerified ?: return BarnardRelayObservationResult.REJECTED
+        if (verified.validFromEnin !in ENIN_RANGE || verified.validThroughEnin !in ENIN_RANGE || verified.relayExpiresAtEnin !in ENIN_RANGE) return BarnardRelayObservationResult.REJECTED
+        if (current < verified.validFromEnin || current >= verified.relayExpiresAtEnin || verified.relayExpiresAtEnin > verified.validThroughEnin || verified.relayExpiresAtEnin - verified.validFromEnin !in 0..12) return BarnardRelayObservationResult.REJECTED
+        candidates[key] = Candidate(digest, envelope, container[1].toInt() and 255, verified.eventId.copyOf(), clock.nowMilliseconds(), verified.validFromEnin, verified.validThroughEnin, verified.relayExpiresAtEnin)
+        selectIfNeeded(selected == null); if (selected == key && (container[1].toInt() and 255) > 0) retain(peerHandle, clock.nowMilliseconds())
+        return BarnardRelayObservationResult.ACCEPTED
+    }
+    public fun advance() {
+        if (stopped) return; val now = clock.nowMilliseconds(); val current = eninSource.currentEnin() ?: run { teardown(); return }
+        candidates.entries.removeIf { !withinRelayWindow(it.value, current) }; if (selected != null && !candidates.containsKey(selected)) deselect(); selectIfNeeded(selected == null || now >= pinUntil)
```

detekt (1.23.8)

[warning] This else block is empty and can be removed.

(detekt.empty-blocks.EmptyElseBlock)